spec.bs

```diff
@@ -1563,3 +1605,135 @@ specification adds another value for fenced frames:
+
+  Rewrite the section to read:
+
+  <{iframe}> and <{fencedframe}> elements have an [=iframe/allow=] attribute, which contains an
+  [=ASCII-serialized policy directive=].
+
+  The allowlist for the features named in the attribute may be empty; in that case, the default
+  value for the allowlist is "`src`", which represents the origin of the URL in the iframe’s src
+  attribute, or the fencedframe's [=fenced frame config=].
+
+  When not empty, the [=iframe/allow=] attribute will result in adding an allowlist for each
+  recognized feature to the iframe element’s content navigable's container policy or the fencedframe
+  element's [=fenced navigable container/fenced navigable=]'s container policy, when it is
+  constructed.
+</div>
+
+<div algorithm=create-permissions-policy-response>
+  Modify the definition of [=Create a Permissions Policy for a navigable from response=] to read:
+
+  Given null, a [=navigable container=], or a [=fenced navigable container=] (|container|), an
+  [=origin=] (|origin|), a [=response=] (<var ignore>response</var>), and null or a
+  list of [=features=] (|required features|), this algorithm returns a new
+  [=Document/permissions policy=].
+
+  Modify step 1 of the algorithm to read:
+  1. Let <var ignore>policy</var> be the result of running
+    [=Create a Permissions Policy for a navigable=] given |container|, |origin|, and
+    |required features|.
+</div>
+
+<div algorithm=shared-document-creation-changes>
+  Modify the [=shared document creation infrastructure=] algorithm. Rewrite step 3 to read:
+
+  3. Let <var ignore>permissionsPolicy</var> be the result of [=Create a Permissions Policy for a
+    navigable from response|creating a permissions policy from a response=] given
+    |navigationParams|'s [=navigable=]'s [=navigable container|container=], |navigationParams|'s
+    [=navigation params/origin=], |navigationParams|'s [=navigation params/response=], and
+    <span class=XXX>TODO: Populate this with the required permissions to load from the config in
+    the navigation params</span>.
+</div>
+
+<div algorithm=new-browsing-context-changes>
+  Modify the [=create a new browsing context and document=] algorithm. Rewrite step 7 to read:
+
+  7. Let <var ignore>permissionsPolicy</var> be the result of [=Create a Permissions Policy for a
+    navigable|creating a permissions policy=] given <var ignore>embedder</var>,
+    <var ignore>origin</var>, and null.
+
+  Note: This change is made in addition to the changes to [=create a new browsing context and
+  document=] outlined in [[#creating-browsing-contexts-patch]].
+</div>
+
+<div algorithm=main-fetch-patches>
+  Modify the [=main fetch=] algorithm. Modify step 19 to add a new bullet point to the algorithm
+  list:
+
+  * [=should response to request be blocked by Permissions Policy=]
+</div>
+
+<div algorithm=permissions-policy-block-request>
+  Create a new algorithm in the [[!permissions-policy]] spec, called <dfn export>should response to
```

Comment:
So this is the algorithm that sees whether or not a fenced frame has all of the permissions that the "config instance" (read: fenced frame properties) demands that it have, right? In that case, I think we should not have this be called by Fetch, but instead basically be one of the many algorithms that we call before we finalize the navigation, that can lead us down the error path. See for example: https://wicg.github.io/fenced-frame/#ref-for-attempt-to-populate-the-history-entry's-document, which basically references algorithms like https://html.spec.whatwg.org/multipage/browsing-the-web.html#attempt-to-populate-the-history-entry's-document:~:text=Otherwise%2C%20if%20the%20result%20of%20checking%20a%20navigation%20response%27s%20adherence%20to%20%60X%2DFrame%2DOptions%60, whose placement I think we want to mimic with the `S-L-M` header checks and probably the permissions policy checks as well, especially since these are navigation-specific checks.